lib/mpl_toolkits/tests/test_mplot3d.py

```diff
@@ -329,6 +329,19 @@ def test_surface3d_shaded():
     ax.set_zlim(-1.01, 1.01)
 
 
+@mpl3d_image_comparison(['surface3d_masked.png'])
+def test_surface3d_masked():
```

Comment:
Does this test exercise both sides of the above `if`?

Reply:
It doesn't and actually now you've pointed it out I realise there is a potential problem here that I didn't consider. If using `rcount/ccount` then you can end up masking all of your data and have no `polys` to draw. An extreme example would be if `rcount=1` and `ccount=1` then all the data would end up masked. Adding a 
```python
if not len(polys):
    raise ValueError(...)
```
with an appropriate error message might be the best solution here, but I'd be interested to hear opinions.

Reply:
`rcount` is really just a drawing optimization; I would ask maybe @eric-wieser or @apaszke for their thoughts.

Reply:
I'm not sure I understand the question. I've used rstride in the past for when I want control of the edge placement on the surface, but never found the API of recount useful.

I do remember changing code using these settings in matplotlib though.

Reply:
Maybe it isn't a problem. The issue is that changing the striding can mean the masked data might be skipped (happy to make a graphical example if needed), which might confuse people. Going the other way, certain `r/c count/stride` values might mean that all the data ends up masked and then no polygons are created and a very cryptic error is raised on drawing: `ValueError: not enough values to unpack (expected 5, got 0)`. I'm just wondering if we should anticipate that and instead raise something like `ValueError: all data masked, no surface will be drawn` when `plot_surface` is called, rather than when drawing.

Reply:
With the latest changes I don't get the `ValueError` anymore for "bad" r/c count/stride values, just an empty plot. I think that is probably fine for now, but let me know if not. I still don't have a test for both sides of the if/else, but it isn't clear to me that any of the other tests for this do. Do you want me to add one?